src/actions/project-text-search.js

```diff
@@ -0,0 +1,55 @@
+// @flow
+
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+/**
+ * Redux actions for the search state
+ * @module actions/search
+ */
+
+import { searchSource } from "../utils/search/project-search";
+
+import { getSources, getSearchResult } from "../selectors";
+
+import { loadAllSources } from "./sources";
+
+import type { ThunkArgs } from "./types";
+
+export function addSearchQuery(query) {
+  return ({ dispatch, getState }: ThunkArgs) => {
+    dispatch({ type: "ADD_QUERY", query });
+  };
+}
+
+export function removeSearchQuery() {
+  return ({ dispatch, getState }: ThunkArgs) => {
+    dispatch({ type: "CLEAR_QUERY" });
+  };
+}
+
+export function searchSources(query) {
+  return async ({ dispatch, getState }: ThunkArgs) => {
+    dispatch(await loadAllSources());
```

Comment:
perhaps we want to do something to disallow new searches while one is going on?

I do this with preview, by having an updating state: https://github.com/devtools-html/debugger.html/blob/master/src/reducers/ast.js#L21-L22